File: app/src/main/java/com/example/worldnewsin/SimpleVideo.java
package com.example.worldnewsin;

import java.text.ParseException;
import java.util.Date;

import com.example.worldnewsin.utils.DateManager;

public class SimpleVideo {
    private Id id;
    private Snippet snippet;

    private final int TITLE_MAX_LENGTH = 90;

    public String getVideoId() {
        return this.id.videoId;
    }

    public String getTitle() {
        String title = this.snippet.title;
        if (title.length() < TITLE_MAX_LENGTH) {
            return title;
        } else {
            return title.substring(0, TITLE_MAX_LENGTH);
        }
    }

    public String getPublishedAt() {
        try {
            return DateManager.formatDate(this.snippet.publishedAt);
        } catch (ParseException e) {
            return this.snippet.publishedAt;
        }
    }

    private class Id {
        private String kind;
        private String videoId;
    }

    private class Snippet {
        private String publishedAt;
        private String title;
    }
}

File: app/src/main/java/online/litterae/worldnewsin/model/SimpleVideo.kt
package online.litterae.worldnewsin.model

import online.litterae.worldnewsin.utils.DateManager
import java.text.ParseException

class SimpleVideo {
    private val id: Id = Id()
    private val snippet: Snippet = Snippet()
    private val TITLE_MAX_LENGTH = 90

    val videoId: String
        get() = id.videoId

    val title: String
        get() {
            val title = snippet.title
            return if (title.length < TITLE_MAX_LENGTH) {
                title
            } else {
                title.substring(0, TITLE_MAX_LENGTH)
            }
        }

    val publishedAt: String
        get() = try {
            DateManager.formatDate(snippet.publishedAt)
        } catch (e: ParseException) {
            snippet.publishedAt
        }

    inner class Id {
        val kind: String = ""
        val videoId: String = ""
    }

    inner class Snippet {
        val publishedAt: String = ""
        val title: String = ""
    }
}